starknet_py/cairo/serialization/data_serializers/payload_serializer.py

```diff
@@ -0,0 +1,85 @@
+from collections import OrderedDict
+from dataclasses import dataclass, InitVar, field
+from typing import Dict, Generator
+
+from starknet_py.cairo.serialization._context import (
+    SerializationContext,
+    DeserializationContext,
+)
+from starknet_py.cairo.serialization.data_serializers._common import (
+    deserialize_to_dict,
+    serialize_from_dict_by_key,
+    serialize_from_dict,
+)
+from starknet_py.cairo.serialization.data_serializers.array_serializer import (
+    ArraySerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.cairo.serialization.data_serializers.felt_serializer import (
+    FeltSerializer,
+)
+
+from starknet_py.cairo.serialization.serialized_payload import SerializedPayload
+
+from starknet_py.utils.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class PayloadSerializer(CairoDataSerializer[Dict, TupleDataclass]):
+    """
+    Serializer for payloads like function arguments/function outputs/events.
+    Can serialize a dictionary.
+    Deserializes data to a TupleDataclass.
+    """
+
+    # Value present only in constructor
+    input_serializers: InitVar[OrderedDict[str, CairoDataSerializer]]
+
+    serializers: OrderedDict[str, CairoDataSerializer] = field(init=False)
+
+    def __post_init__(self, input_serializers):
+        """
+        ABI adds ARG_len for every argument ARG that is an array. We parse length as a part of ArraySerializer, so we
+        need to remove those lengths from args.
+        """
+        self.serializers = OrderedDict(
+            (key, serializer)
+            for key, serializer in input_serializers.items()
+            if not self._is_len_arg(key, input_serializers)
+        )
+
+    # This is added because PreparedFunctionCall has a dict with every member serialized.
+    def serialize_members(self, members: Dict) -> SerializedPayload:
+        """
+        Method accepting members as a dictionary and serializing them separately.
+
+        :return: Members serialized separately in SerializedPayload.
+        """
+        with SerializationContext.create() as context:
+            return SerializedPayload(
+                (name, list(values))
+                for name, values in serialize_from_dict_by_key(
+                    self.serializers, context, members
+                )
+            )
```

Comment:
I'm not sure if I understand possible use cases for this method. Could you elaborate on that?

Reply:
The only reason I added it is: it was previously supported and exposed in PreparedFunctionCall. It complicates everything unfortunately. 

Now that I think about it, perhaps we should deprecate it there before the new data transformer is added and just ignore it completely? I hate that it adds more complexity for amost no gain. I didn't plan to add this at the beginning, but when I started integrating with contract.py I noticed it is missing. @war-in @drknzz please provide your view on this as well.

Reply:
I remove serialize_members from PR and created https://github.com/software-mansion/starknet.py/pull/610.